src/Metrics/MetricsCalculator.php

```diff
@@ -44,28 +43,14 @@
 /**
  * @internal
  */
-class MetricsCalculator
+class MetricsCalculator implements Collector
 {
     private int $roundingPrecision;
-    private SortableMutantExecutionResults $killedExecutionResults;
-    private SortableMutantExecutionResults $errorExecutionResults;
-    private SortableMutantExecutionResults $escapedExecutionResults;
-    private SortableMutantExecutionResults $timedOutExecutionResults;
-    private SortableMutantExecutionResults $skippedExecutionResults;
-    private SortableMutantExecutionResults $notCoveredExecutionResults;
-    private SortableMutantExecutionResults $allExecutionResults;
 
-    private int $killedCount = 0;
-
-    private int $errorCount = 0;
-
-    private int $skippedCount = 0;
-
-    private int $escapedCount = 0;
-
-    private int $timedOutCount = 0;
-
-    private int $notCoveredByTestsCount = 0;
+    /**
+     * @var array<string, int>
+     */
+    private array $countByStatus = [];
```

Comment:
Contrary to the above, the array-key type is significant here. We'll be using it and comparing against strings.